cluster_management/src/main/java/com/pinterest/rocksplicator/utils/PartitionStateManager.java

```diff
@@ -0,0 +1,131 @@
+/// Copyright 2021 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author prem (prem@pinterest.com)
+//
+
+package com.pinterest.rocksplicator.utils;
+
+import org.apache.curator.RetryLoop;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.KeeperException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class PartitionStateManager {
+    private static final Logger LOG = LoggerFactory.getLogger(PartitionStateManager.class);
+    private static final String LOCAL_HOST_IP = "127.0.0.1";
+    final CuratorFramework zkClient;
+    final String cluster;
+    final String resourceName;
+    final String partitionName;
+    final PartitionState oldState;
+
+    public PartitionStateManager (
+      final String cluster,
+      final String resourceName,
+      final String partitionName,
+      final CuratorFramework zkClient) {
+        this.cluster = cluster;
+        this.resourceName = resourceName;
+        this.partitionName = partitionName;
+        this.zkClient = zkClient;
+        oldState = new PartitionState();
+    }
+
+    public String getStatePath() {
+      return String.format("/pstate/%s/%s/%s",cluster,resourceName,partitionName);
```

Comment:
maybe use `partitionstate` in the zk path to make it clear

Reply:
Done